earthly

```diff
@@ -124,7 +136,11 @@ case "$1" in
 
         last_flag_hash=$(cat "$last_flag_hash_path" 2>/dev/null || echo null)
         flagoverride_hash=$(md5sum "$earthly_version_flag_overrides_path" | awk '{print $1}')
-        if [ -z "$COMP_LINE" ]; then
+        if [ ! -x "$bindir/earthly-prerelease" ]; then
+            echo "Could not find an executable '$bindir/earthly-prerelease'."
```

Comment:
```suggestion
            echo "Could not find an executable '$bindir/earthly-prerelease'." >&2
```

Reply:
Could you move this entire check under the 

```
if [ -z "$COMP_LINE" ]; then

...
elif [ "$since" -ge 3600 ]; then
   ...
elif [ ! -x "$bindir/earthly-prerelease" ]; then
   echo "Could not find an executable '$bindir/earthly-prerelease'."
            update="true"
            get_latest_binary_and_update_state

```

block?

I want to avoid any potential updates when resolving auto-completion -- I would rather have earthly fail-fast (and not potentially print anything to stdout)

Reply:
That won't work since the normal case we care about will be `[ -z "$COMP_LINE" ]`.

For autocompletion, we could do something like:
```sh
if [ ! -x "$bindir/earthly-prerelease" ]; then
  if [ -z "$COMP_LINE" ]; then
     echo "Could not find an executable '$bindir/earthly-prerelease'." >&2
              update="true"
              get_latest_binary_and_update_state
    ...
  else
    # handle autocompletion when there's no binary...
  fi
elif [ -z "$COMP_LINE" ]; then

...
elif [ "$since" -ge 3600 ]; then
   ...
```

Reply:
Here's more or less what I was suggesting: #2356